data-integration/main.bal

```diff
@@ -0,0 +1,103 @@
+import ballerina/http;
+import ballerina/os;
+import ballerina/persist;
+import ballerina/lang.runtime;
+import data_integration.db;
+
+configurable int port = 9090;
+
+service /employees on new http:Listener(port) {
+    private final db:Client dbClient;
+
+    function init() returns error? {
+        _ = check startDBService();
+        self.dbClient  = check new();
+    }
+
+    isolated resource function post employees(db:EmployeeInsert employee)
+            returns http:InternalServerError|http:Created|http:Conflict {
+        int[]|persist:Error result = self.dbClient->/employees.post([employee]);
+        if result is persist:Error {
+            if result is persist:AlreadyExistsError {
+                return http:CONFLICT;
+            }
+            return http:INTERNAL_SERVER_ERROR;
+        }
+        return http:CREATED;
+    }
+
+    isolated resource function post tasks(db:EmployeeTaskInsert task)
+            returns http:InternalServerError|http:Created|http:Conflict {
+        int[]|persist:Error result = self.dbClient->/employeetasks.post([task]);
+        if result is persist:Error {
+            if result is persist:AlreadyExistsError {
+                return http:CONFLICT;
+            }
+            return http:INTERNAL_SERVER_ERROR;
+        }
+        return http:CREATED;
+    }
+
+    isolated resource function get tasks(int taskId) returns db:EmployeeTask[]|error {
```

Comment:
We shouldn't return `error`. We need to handle errors and send an appropriate response.